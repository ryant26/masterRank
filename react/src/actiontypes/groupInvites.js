export const ADD_GROUP_INVITE = 'groupInvites/ADD_GROUP_INVITE';