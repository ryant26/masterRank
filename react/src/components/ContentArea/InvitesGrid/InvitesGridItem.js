import React from 'react';
import PropTypes from 'prop-types';
import HeroCard from '../DashboardHome/HeroRoles/HeroCard/HeroCard';

const InvitesGridItem = ({invite}) => {
    let numberOfMembers = invite.members.length + 1;
    let baseSr = invite.leader.skillRating;
    baseSr = invite.members.reduce((sr, member) => sr + member.skillRating, baseSr);
    let avgSrString = (baseSr/numberOfMembers).toFixed(0);

    let groupHeroes = [<HeroCard key={invite.leader.platformDisplayName} user={invite.leader} hero={invite.leader}/>];
    invite.members.forEach((member) => {
        groupHeroes.push(<HeroCard key={member.platformDisplayName} user={member} hero={member}/>);
    });

    return (
        <div className="InvitesGridItem flex flex-column card">
            <div className="header">
                <div className="group-title flex align-center justify-center">
                    <h3>{invite.leader.platformDisplayName}'s</h3>
                    <h3 className="regular">Group</h3>
                </div>
                <div className="groupSR sub-title flex align-center justify-center">
                    {avgSrString} Group SR
                </div>
            </div>
            <div className="body flex flex-column">{groupHeroes}</div>
            <div className="button-group flex justify-around">
                <div className="button button-five flex justify-center align-center">
                    <div className="button-content">
                        DECLINE
                    </div>
                </div>
                <div className="button button-secondary flex justify-center align-center">
                    <div className="button-content">
                        ACCEPT
                    </div>
                </div>
            </div>
        </div>
    );
};

InvitesGridItem.propTypes = {
    invite: PropTypes.object.isRequired
};

export default InvitesGridItem;