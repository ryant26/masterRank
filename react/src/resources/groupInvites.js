const groupInvites = [
  {
    groupId: 10,
    groupSize: 4,
    inviteDate: "2018-02-19T12:57:24.790Z",
    leader: {
      platformDisplayName: "luckbomb#1234",
      skillRating: 3200,
      heroName: "genji",
      stats: {}
    },
    members: [
      {
        platformDisplayName: "wismo0#1234",
        skillRating: 2200,
        heroName: "widowmaker",
        stats: {}
      }
    ],
    pending: [
      {
        platformDisplayName: "andyLin#1234",
        skillRating: 1800,
        heroName: "tracer",
        stats: {}
      }
    ]
  },
  {
    groupId: 8,
    groupSize: 3,
    inviteDate: "2017-09-19T21:57:24.790Z",
    leader: {
      platformDisplayName: "luckbomb#1234",
      skillRating: 3200,
      heroName: "genji",
      stats: {}
    },
    members: [],
    pending: [
      {
        platformDisplayName: "cutie#1320",
        skillRating: 1600,
        heroName: "tracer",
        stats: {}
      }
    ]
  },
    {
        groupId: 2,
        groupSize: 4,
        inviteDate: "2017-06-19T16:57:24.790Z",
        leader: {
            platformDisplayName: "luckbomb#1234",
            skillRating: 3200,
            heroName: "doomfist",
            stats: {}
        },
        members: [
            {
                platformDisplayName: "wismo0#1234",
                skillRating: 2200,
                heroName: "soldier76",
                stats: {}
            }
        ],
        pending: [
            {
                platformDisplayName: "cutie#1320",
                skillrating: 1600,
                heroName: "winston",
                stats: {}
            }
        ]
    },
     {
      groupId: 210,
      groupSize: 6,
      inviteDate: "2018-02-19T16:57:24.790Z",
      leader: {
        platformDisplayName: "luckbomb#1234",
        skillRating: 3200,
        heroName: "genji",
        stats: {}
      },
      members: [
        {
          platformDisplayName: "wismo0#1234",
          skillRating: 2200,
          heroName: "widowmaker",
          stats: {}
        },{
          platformDisplayName: "andyLin#1234",
          skillRating: 1600,
          heroName: "winston",
          stats: {}
        }
      ],
      pending: [
        {
          platformDisplayName: "cutie#1320",
          skillRating: 1800,
          heroName: "tracer",
          stats: {}
        },{
          platformDisplayName: "pwnShopp#4810",
          skillRating: 1800,
          heroName: "soldier76",
          stats: {}
        },{
          platformDisplayName: "banannaTruck#1872",
          skillRating: 1800,
          heroName: "doomfist",
          stats: {}
        }
      ]
    },
];

export default groupInvites;