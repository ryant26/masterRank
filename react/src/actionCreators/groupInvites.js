import * as GroupInviteActionTypes from 'actiontypes/groupInvites';

export const addGroupInvite = (invite) => {
  return {
    type: GroupInviteActionTypes.ADD_GROUP_INVITE,
    invite
  };
};

export const removeGroupInvite = (invite) => {
  return {
    type: GroupInviteActionTypes.REMOVE_GROUP_INVITE,
    invite
  };
};