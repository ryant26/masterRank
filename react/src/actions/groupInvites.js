import * as GroupInviteActionTypes from '../actiontypes/groupInvites';

export const addGroupInvite = (invite) => {
  return {
    type: GroupInviteActionTypes.ADD_GROUP_INVITE,
    invite
  };
};