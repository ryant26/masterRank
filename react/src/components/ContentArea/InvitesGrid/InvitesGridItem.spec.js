import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import InvitesGridItem from './InvitesGridItem';
import Model from '../../../model/model';
import groupInvites from '../../../resources/groupInvites';


const mockStore = configureStore();
const getHeroCardComponent = (group) => {
    let store = mockStore({
        group: group,
    });

    return shallow(
        <InvitesGridItem invite={group} store={store}/>
    );
};

describe('InvitesGridItem Component', () => {
    const group = groupInvites[0];

    it('should match the snapshot', () => {
        let store = mockStore({
            group: group,
        });
        const component = renderer.create(
            <Provider  store={store}>
                <InvitesGridItem invite={group}/>
            </Provider>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render without exploding', () => {
        const wrapper = getHeroCardComponent(group);
        const InvitesGridComponent = wrapper.find(InvitesGridItem);
        expect(InvitesGridComponent).toBeTruthy();
    });

    it('should show the correct group SR', () => {
        const wrapper = getHeroCardComponent(group);
        expect(wrapper.find('.groupSR').text()).toEqual('2700 Group SR');
    });

    it('should show SR without the comma when < 1000', () => {
        const groupInvite = {
            groupId: 10,
            groupSize: 2,
            inviteDate: new Date(),
            leader: {
                platformDisplayName: "luckbomb#1234",
                skillRating: 900,
                heroName: "genji",
                stats: {}
            },
            members: [],
            pending: []
        };

        const wrapper = getHeroCardComponent(groupInvite);
        expect(wrapper.find('.groupSR').text()).toEqual('900 Group SR');
    });

    describe('when accept button is clicked', () => {
        let HeroCardComponent;

        beforeEach(() => {
            Model.leaveGroup = jest.fn();
            Model.acceptInvite = jest.fn();
            HeroCardComponent = getHeroCardComponent(group);
            HeroCardComponent.find('.button-secondary').simulate('click');
        });

        it('should call Model.leaveGroup with group id', () => {
            expect(Model.leaveGroup).toHaveBeenCalledWith(group.groupId);
        });

        it('should call Model.acceptInvite with group id', () => {
            expect(Model.acceptInvite).toHaveBeenCalledWith(group.groupId);
        });
    });

    it('should call Model.declineInvite when decline button is clicked', () => {
        Model.declineInvite = jest.fn();
        const wrapper = getHeroCardComponent(group);
        wrapper.find('.button-six').simulate('click');
        expect(Model.declineInvite).toHaveBeenCalledWith(group.groupId);
    });
});