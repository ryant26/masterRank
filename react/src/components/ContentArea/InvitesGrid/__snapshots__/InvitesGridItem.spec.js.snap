// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InvitesGridItem Component should match the snapshot 1`] = `
<div
  className="InvitesGridItem flex flex-column card"
>
  <div
    className="header"
  >
    <div
      className="group-title flex align-center justify-center"
    >
      <h3>
        luckbomb#1234
        's
      </h3>
      <h3
        className="regular"
      >
        Group
      </h3>
    </div>
    <div
      className="groupSR sub-title flex align-center justify-center"
    >
      2700
       Group SR
    </div>
  </div>
  <div
    className="body flex flex-column"
  >
    <div
      className="HeroCard flex align-center"
    >
      <div
        className="invitePlayerButton"
      >
        <img
          alt="genji icon"
          className="HeroImage"
          onClick={undefined}
          src={
            Object {
              "process": [Function],
            }
          }
        />
        <div
          className="overlay"
        >
          <div
            className="plus-container flex align-center justify-center"
            onClick={[Function]}
          >
            <i
              className="fa fa-plus"
            />
          </div>
        </div>
      </div>
      <div
        className="content flex flex-column"
      >
        <div
          className="display-name"
        >
          luckbomb#1234
        </div>
        <div
          className="sub-title"
        >
          <span>
            3200
             SR
          </span>
          <span>
             | 
          </span>
          <span>
            0
            % WR
          </span>
        </div>
      </div>
      <div
        className="button-primary"
        onClick={[Function]}
      >
        <div
          className="button-content"
        >
          stats
        </div>
      </div>
    </div>
    <div
      className="HeroCard flex align-center"
    >
      <div
        className="invitePlayerButton"
      >
        <img
          alt="widowmaker icon"
          className="HeroImage"
          onClick={undefined}
          src={
            Object {
              "process": [Function],
            }
          }
        />
        <div
          className="overlay"
        >
          <div
            className="plus-container flex align-center justify-center"
            onClick={[Function]}
          >
            <i
              className="fa fa-plus"
            />
          </div>
        </div>
      </div>
      <div
        className="content flex flex-column"
      >
        <div
          className="display-name"
        >
          wismo0#1234
        </div>
        <div
          className="sub-title"
        >
          <span>
            2200
             SR
          </span>
          <span>
             | 
          </span>
          <span>
            0
            % WR
          </span>
        </div>
      </div>
      <div
        className="button-primary"
        onClick={[Function]}
      >
        <div
          className="button-content"
        >
          stats
        </div>
      </div>
    </div>
    <div
      className="HeroCard flex align-center"
    >
      <div
        className="invitePlayerButton"
      >
        <img
          alt="tracer icon"
          className="HeroImage"
          onClick={undefined}
          src={
            Object {
              "process": [Function],
            }
          }
        />
        <div
          className="overlay"
        >
          <div
            className="plus-container flex align-center justify-center"
            onClick={[Function]}
          >
            <i
              className="fa fa-plus"
            />
          </div>
        </div>
      </div>
      <div
        className="content flex flex-column"
      >
        <div
          className="display-name"
        >
          andyLin#1234
        </div>
        <div
          className="sub-title"
        >
          <span>
            1800
             SR
          </span>
          <span>
             | 
          </span>
          <span>
            0
            % WR
          </span>
        </div>
      </div>
      <div
        className="button-primary"
        onClick={[Function]}
      >
        <div
          className="button-content"
        >
          stats
        </div>
      </div>
    </div>
  </div>
  <div
    className="button-group flex justify-around"
  >
    <div
      className="button button-six flex justify-center align-center"
      onClick={[Function]}
    >
      <div
        className="button-content"
      >
        DECLINE
      </div>
    </div>
    <div
      className="button button-secondary flex justify-center align-center"
      onClick={[Function]}
    >
      <div
        className="button-content"
      >
        ACCEPT
      </div>
    </div>
  </div>
</div>
`;
